sql3/planner/inbuiltfunctionsdate.go

```diff
@@ -196,3 +220,82 @@ func (n *callPlanExpression) EvaluateToTimestamp(currentRow []interface{}) (inte
 	//should we throw error or return nil if the conversion fails? what is the desired behaviour when ToTimestamp errors for one bad record in a batch of thousands?
 	return featurebase.ValToTimestamp(unit, num)
 }
+
+func (n *callPlanExpression) EvaluateDateTimeName(currentRow []interface{}) (interface{}, error) {
+	intervalEval, err := n.args[0].Evaluate(currentRow)
+	if err != nil {
+		return nil, err
+	}
+
+	dateEval, err := n.args[1].Evaluate(currentRow)
+	if err != nil {
+		return nil, err
+	}
+
+	// nil if anything is nil
+	if intervalEval == nil || dateEval == nil {
```

Comment:
Can/should this happen? I don't fully understand this, but I guess I'm wondering whether either of those should be an error, rather than a non-error nil return.

Reply:
Reading it more: I think if intervalEval is nil, that should be an error, probably. I think it may make sense to allow nulls and assert that DateTimeName('m', NULL) is NULL, because that's sort of how SQL rolls with nulls, but I think DateTimeName(NULL, date) should be an error, probably?

Reply:
`fbsql> select datepart('m', '2022-10-11T12:31:45.234Z');

----
 10

fbsql> select datepart(NULL, '2022-10-11T12:31:45.234Z');

------
 NULL

fbsql> select datepart('m', NULL);

------
 NULL`
 it looks like it can happen. I think it's debatable whether or not it should be an error? it at least seems consistent to me that if you try to get a NULL part of a timestamp you get NULL back, but possibly an error would be more useful?